### PR TITLE
CORE-7001: Define notary lookup service

### PR DESCRIPTION
**Pull request by yift-r3**
Runtime PR in https://github.com/corda/corda-runtime-os/pull/2348

### CODE REVIEW

File: application/src/main/kotlin/net/corda/v5/application/membership/NotaryLookup.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+     * ```
+     * list.joinToString("|")
+     * list.map { "[$it]" }.toString()
+     * ```
+     * This allows to distinguish between each element of a map, as
+     * [MemberX500Name.toString] returns comma separated values of [MemberX500Name].
+     *
+     * @return A list of the network's Notary services.
+     */
+    @Suspendable
```
Comment (charlieR3) [inline application/src/main/kotlin/net/corda/v5/application/membership/NotaryLookup.kt]: I'm curious why is this required?

**Review by relyafi — COMMENTED** [review]
LGTM

File: application/src/main/kotlin/net/corda/v5/application/membership/NotaryLookup.kt (hunk 1 above)
Comment (charlieR3) [reply application/src/main/kotlin/net/corda/v5/application/membership/NotaryLookup.kt]: It seem like this is required on all functions based on the conversations in the runtime os PR

Comment (charlieR3) [reply application/src/main/kotlin/net/corda/v5/application/membership/NotaryLookup.kt]: Added

**Comment by charlieR3** [comment]
Discusses with Matthew at today's standup. This may now be a ledger concern that we do not need to work on. Waiting until Christian is back from PTO on Thursday to confirm.

**Comment by charlieR3** [comment]
> Discusses with Matthew at today's standup. This may now be a ledger concern that we do not need to work on. Waiting until Christian is back from PTO on Thursday to confirm.

Agreed that we would continue this work. `NotaryInfo` and `NotaryLookup` can be moved to the ledger common module. This post seems relevant https://r3holdco.slack.com/archives/C03A3TQMBHR/p1667380816659219

**Review by charlieR3 — DISMISSED** [review]

**Review by charlieR3 — APPROVED** [review]
